### PR TITLE
[Feature] Support qualitative comparison tools

### PR DESCRIPTION
**Pull request by Z-Fran**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Provide qualitative comparison tools in MMEditing.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

python tools/gui/gui.py

## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmediting/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [x] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [x] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects.
- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by zengyh1900** [comment]
Please provide a detailed use case (e.g., the script to run) in the description.

**Comment by zengyh1900** [comment]
You may also need to update the requirement of 'PyQt5' in requirements/vis.txt

File: tools/gui/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
+# MMEditing Viewer
+
+## Introduction
+
+**MMEditing Viewer** is a qualitative comparison tools to facilitate your research.
+
```
Comment (zengyh1900) [inline tools/gui/README.md:6]: You can add an overview of the content here.

File: tools/gui/README.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+- **Patch-based comparison**
+  - Crop a patch on multiple images to compare
+  - Batch comparison
+  - Flexible settings on number of columns and size of images.
+  - Save your comparison result
+- **Before/After slider comparison**
+  - Support both videos and images comparison
+  - Record and save comparison results as a video clip
+
```
Comment (zengyh1900) [inline tools/gui/README.md:17]: You may need add a section about environment/prerequirements/platform to introduce:
supported python version, etc.

**Comment by zengyh1900** [comment]
Please check and delete the saved cache images.

**Comment by Z-Fran** [comment]
> Please check and delete the saved cache images.

**Review by zengyh1900 — APPROVED** [review]
Good job! You have made such an amazing tool that would help many users visualize and compare! I like this tool! 💘💘💘

**Pull request merged**